README.md

```diff
@@ -24,6 +24,381 @@ its own version allows changes to be made to the client outside of updates to th
+export CONJUR_SSL_CERTIFICATE="<Certificate Text>"
+```
+
+#### Setting Certificate Programmatically
+
+The API client can be configured by using either the `setCertFile`, or `setSslCaCert`
+methods if you have a filepath or certificate text respectively.
+
+```java
+import com.cyberark.conjur.sdk.Configuration;
+import com.cyberark.conjur.sdk.ApiClient;
+
+// Get the systems default API client
+ApiClient client = Configuration.getDefaultApiClient();
+
+client.setCertFile("<filepath>");
+// or
+client.setSslCaCert(CERT_INPUT_STREAM);
+
+// If we want to make sure this API client configuration is always used
+Configuration.setDefaultApiClient(client);
+```
+
+### Authorization Examples
+
+As mentioned in the [Configuration](#configuration) section, you can provide varying ways
+for your app to authenticate against a Conjur server. Environment variables are the best
+way to do this as authentication will all occur behind the scenes. However you can set these
+values manually. Once you have chosen from one of the patterns below that works for you, you
+can now create a `Conjur` class instance values to access Conjur services and make RESTful API calls.
+
+#### Environment Variables
+
+```bash
+export CONJUR_ACCOUNT=<account specified during Conjur setup>
+export CONJUR_APPLIANCE_URL=<Conjur endpoint URL>
+export CONJUR_AUTHN_LOGIN=<user/host identity>
+export CONJUR_AUTHN_API_KEY=<user/host API key (not password) - see notes about `CONJUR_AUTHN_URL`>
+export CONJUR_CERT_FILE=<path to ca.crt file>
+```
+```java
+import com.cyberark.conjur.sdk.endpoint.SecretsApi;
+
+// Configured/authenticated using environment variables
+SecretsApi api = new SecretsApi();
+```
+
+#### Username and API key
+
+```bash
+export CONJUR_ACCOUNT=<account specified during Conjur setup>
+export CONJUR_APPLIANCE_URL=<Conjur endpoint URL>
+export CONJUR_CERT_FILE=<path to ca.crt file>
+```
+
+```java
+import com.cyberark.conjur.sdk.Configuration;
+import com.cyberark.conjur.sdk.ApiClient;
+
+// Get the systems default API client
+ApiClient client = Configuration.getDefaultApiClient();
+// Given you already have an api key you can just set the clients value
+client.setUsername("username");
+client.setApiKey("API key");
+
+// If we want to make sure this API client configuration is always used
+Configuration.setDefaultApiClient(client);
+```
+
+#### Username and Password
+
+_Note:_ This is only for non host users.
+
+```bash
+export CONJUR_ACCOUNT=<account specified during Conjur setup>
+export CONJUR_APPLIANCE_URL=<Conjur endpoint URL>
+export CONJUR_CERT_FILE=<path to ca.crt file>
+```
+
+```java
+import com.cyberark.conjur.sdk.ApiClient;
+import com.cyberark.conjur.sdk.Configuration;
+import com.cyberark.conjur.sdk.endpoint.AuthenticationApi;
+
+// Get the systems default API client
+ApiClient client = Configuration.getDefaultApiClient();
+// Given you already have an api key you can just set the clients value
+client.setUsername("username");
```

Comment:
came across this PR and just popping in here for a second :)
Is this sustainable if we keep adding new ways to login to the client? 

What about items like account, cert_file, and url. Can they be set in this way as well?